docs-go/cloud-run.md

```diff
@@ -0,0 +1,174 @@
+<!-- Autogenerated by weave; DO NOT EDIT -->
+
+# Firebase Genkit with Cloud Run
+
+You can deploy Firebase Genkit flows as web services using Cloud Run. This page,
+as an example, walks you through the process of deploying the default sample
+flow.
+
+1.  Install the [Google Cloud CLI](https://cloud.google.com/sdk/docs/install) if
+    you haven't already.
+
+1.  Create a new Google Cloud project using the
+    [Cloud console](https://console.cloud.google.com) or choose an existing one.
+    The project must be linked to a billing account.
+
+    After you create or choose a project, configure the Google Cloud CLI to use
+    it:
+
+    ```posix-terminal
+    gcloud auth login
+
+    gcloud init
+    ```
+
+1.  Create a directory for the Genkit sample project:
+
+    ```posix-terminal
+    mkdir -p ~/tmp/genkit-cloud-project
+
+    cd ~/tmp/genkit-cloud-project
+    ```
+
+    If you're going to use an IDE, open it to this directory.
+
+1.  Initialize a Go module in your project directory:
+
+    ```posix-terminal
+    go mod init example/cloudrun
+    ```
+
+1.  Initialize Genkit in your project:
+
+    ```posix-terminal
+    genkit init
+    ```
+
+    Select the model provider you want to use.
+
+    Accept the defaults for the remaining prompts. The `genkit` tool will create
+    a sample source file to get you started developing your own AI flows.
+    For the rest of this tutorial, however, you'll just deploy the sample flow.
+
+1.  Edit the sample file (`main.go` or `genkit.go`) to explicitly specify the
+    port the flow server should listen on:
+
+    ```go
+    if err := genkit.Init(ctx,
+    	&genkit.Options{FlowAddr: ":3400"}, // Add this parameter.
```

Comment:
I think this happens by default.

Reply:
The default is `127.0.0.1:3400`, which doesn't work when deployed as far as I can tell.

Reply:
It should first try to get the port number from PORT env var: https://github.com/firebase/genkit/blob/4b36ece6177cbd61ffa319d8fbde987c21d27e6e/go/genkit/servers.go#L68

Reply:
Yeah,. but it still prepends 127.0.0.1:

https://github.com/firebase/genkit/blob/4b36ece6177cbd61ffa319d8fbde987c21d27e6e/go/genkit/servers.go#L362-L364

Reply:
Oh.... that's a bug.